ShoppingCart/ShoppingCart/CartItem.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace ShoppingCart
+{
+    public class CartItem
+    {
+        public Product item { get; private set; }
+        public double Quantity { get; private set; }
+        public double TotalItemCost { get; private set; }
+
+        public CartItem(Product product, double quantity)
+        {
+            item = new Product(product.Name,product.PricePerUnit);

```

Comment:
Unnecessary creation of new object. item = product; should move the reference to the item object.